Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1094,6 +1094,13 @@ open class NavigationMapView: UIView {
             guard let self = self else { return }
             let point = self.mapView.mapboxMap.point(for: location.coordinate)
             userCourseView.center = point
+            
+            // If the point is outside of the bounds of `MapView` - hide user course view.
+            if point.x == -1.0 && point.y == -1.0 {
```

Comment:
I would say that we don't need to set `userCourseView.center = point` in this case, no? And it's not clear if we need animation here.

Reply:
I agree that this check should happen as soon as possible (moved it to be executed in first place). At the same time I do not like such fix at all, but by the looks of it there is not much SDK can do at this point.